controllers/trainingjob/trainingjob_controller.go

```diff
@@ -185,8 +191,24 @@ func (r *Reconciler) reconcileTrainingJob(ctx reconcileRequestContext) error {
 		break
 
 	case sagemaker.TrainingJobStatusCompleted:
-		if err = r.addModelPathToStatus(ctx); err != nil {
-			return r.updateStatusAndReturnError(ctx, ReconcilingTrainingJobStatus, "", errors.Wrap(err, "Unable to add model path to status"))
+		if len(ctx.TrainingJob.Status.ModelPath) == 0 {
+			if err = r.addModelPathToStatus(ctx); err != nil {
+				return r.updateStatusAndReturnError(ctx, ReconcilingTrainingJobStatus, "", errors.Wrap(err, "Unable to add model path to status"))
+			}
+		}
+		// debug rule runs asynchronously with trainingjob. In some cases debug rule
```

Comment:
nit: Can we start off comments with proper sentence grammar? Things like capitalizing the first word helps readability a lot.
```
Debug rules run asynchronously with training jobs. (...)
```